test/util_tests/string_format.cpp

```diff
@@ -0,0 +1,65 @@
+#include "common/string_format.h"
```

Comment:
We have stopped adding unit tests now since they are hard to maintain.
I will let @andyfengHKU comment on this

Reply:
I will push strongly for unit tests for a custom formatter. This is a perfectly self-contained unit test.

I understand why we stopped adding unit tests for other things that can be easily tested through E2E tests, but this cannot.

Reply:
Yes this type of unit test is fine.

I was against adding unit test for operator, functions, etc.